### DISCUSSION TITLE
Renaming core stuff (left, top to x, y) and breaking changes

### DISCUSSION THREAD

**Discussion by ShaMan123** [fabricjs/fabric.js#7500] [General]
I think there are some parts in fabric that need to be renamed.
`left`, `top` is the perfect example.
For a dev starting with fabric it makes no sense at all (for anyone actually), it should have been renamed when `originX`, `originY` were included.
It feels bad to have too many quirks to manage. Gives a general feeling of a mess and instability.
I think for the sake of the future of fabric renaming and breaking changes must happen, better sooner than later.

@asturur ??

**Comment by asturur** [comment]
Ideally yes, but can you imagine the mess? With the speed this project is going, those are easy 50 issues opened in a week from release.
if you ask me, originX and originY set to center and no options are good enough.
With the current team this is really low priority.
